### PR TITLE
Return available meeting times for mandatory + optional members

### PR DESCRIPTION
**Pull request by swalsh15**
Algorithm to find available meeting times for list of mandatory and optional members given length of the meeting and other meetings scheduled for that day. Pass 21 tests given with file and 5 tests I wrote to verify optional members are added correctly. This is from the week 5 walkthrough.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,98 @@
...
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: Can we break this up into a few smaller functions for readability?

**Review by augerm — CHANGES_REQUESTED** [review]

**Review by Danish21 — APPROVED** [review]
Nice job on this. I like the clarity of your tests!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,126 @@
...
+    } else {
+      mandatoryMembers = request.getAttendees();
+      optionalMembers = request.getOptionalAttendees(); 
+    }
+
+    // go through all events and find unavailable times
+    ArrayList<TimeRange> unavailableTimes = findUnavailableTimesMandatoryAttendees(events, mandatoryMembers);   
+
+    // find available times from unavilable times from mandatory members
+    ArrayList<TimeRange> availableTimes = findAvalibaleTimesMandatoryMembers(unavailableTimes, request);
```
Comment (augerm) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:42]: `findAvalibaleTimesMandatoryMembers` --> `findAvailableTImesMandatoryMembers` typo in available

**Review by augerm — APPROVED** [review]
Thanks for the changes!

**Pull request merged**